spec/features/navigation/navigation_spec.rb

```diff
@@ -71,9 +58,126 @@
+
+      fill_in :email, with: kiera.email
+      fill_in :password, with: kiera.password
+
+      click_button 'Login'
+
+      within 'nav' do
+        expect(page).to have_link("#{kiera.name}'s Profile")
+        expect(page).to have_link("Logout")
+        expect(page).to_not have_link("User Log In")
+        expect(page).to_not have_link("Register New User")
+        expect(page).to have_content("Logged in as #{kiera.name}")
+      end
+    end
+  end
+
+  describe 'As a Merchant' do
+    it "shows same links as users plus dashboard link" do
+      kiera = User.create!(name: 'Kiera Allen', address: '124 Main St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@marley.com', password: 'password', role: 1)
+
+      visit '/login'
+
+      fill_in :email, with: kiera.email
+      fill_in :password, with: kiera.password
+
+      click_button 'Login'
+
+      within 'nav' do
+        expect(page).to have_link("Merchant's Dashboard")
+      end
+    end
+  end
+
+  describe 'As a Admin' do
+    it "shows same links as users plus dashboard link and all users" do
+      kiera = User.create!(name: 'Kiera Allen', address: '124 Main St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@marley.com', password: 'password', role: 2)
+
+      visit '/login'
+
+      fill_in :email, with: kiera.email
+      fill_in :password, with: kiera.password
+
+      click_button 'Login'
+
+      within 'nav' do
+        expect(page).to have_link("Admin's Dashboard")
+        expect(page).to have_link("All Users")
+      end
+    end
+  end
+
+  describe 'As a visitor' do
+    it "When I try to access any path that begins with the following, then I see a 404 error:" do
+
+      visit "/merchant"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+
+      visit "/admin"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+
+      visit "/profile"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+    end
+  end
+
+  describe 'As a user' do
+    it "When I try to access any path that begins with the following, then I see a 404 error:" do
+      kiera = User.create!(name: 'Kiera Allen', address: '124 Main St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@marley.com', password: 'password')
+
+      visit '/login'
+
+      fill_in :email, with: kiera.email
+      fill_in :password, with: kiera.password
+
+      click_button 'Login'
+
+      visit "/merchant"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+
+      visit "/admin"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+    end
+  end
+
+  describe 'As a merchant' do
+    it "When I try to access any path that begins with the following, then I see a 404 error:" do
+      kiera = User.create!(name: 'Kiera Allen', address: '124 Main St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@marley.com', password: 'password', role: 1)
```

Comment:
Great job! We may want to stick with different user names for the different roles in our tests so it's easier to tell the difference. For example, in the session new spec we have three different names/info.

for merchant:
`sally = User.create!(name: 'Sally Peach', address: '432 Grove St.', city: 'Denver', state: 'CO', zip: 80205, email: 'sally@peach.com', password: 'password', role: 1)`

for admin:
`bob = User.create!(name: 'Bob Ross', address: '745 Rose St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@ross.com', password: 'password', role: 2)`